storage/grpc_client.go

```diff
@@ -783,7 +783,7 @@ func (c *grpcStorageClient) RewriteObject(ctx context.Context, req *rewriteObjec
 	if s.userProject != "" {
 		ctx = setUserProjectMetadata(ctx, s.userProject)
 	}
-	if err := applyCondsProto("Copy destination", req.gen, req.conds, call); err != nil {
+	if err := applyCondsProto("Copy destination", req.srcObject.gen, req.dstObject.conds, call); err != nil {
```

Comment:
This is still not correct -- you should just pass -1 for generation to indicate the latest. Passing a specific generation for a destination object is not possible.

Reply:
Yup looks good now! Thanks!